include/fmt/core.h

```diff
@@ -987,7 +988,13 @@ class format_arg_store {
   friend class basic_format_args<Context>;
 
  public:
+#if FMT_USE_CONSTEXPR
+  static constexpr uint64_t TYPES =
+      IS_PACKED ? internal::get_types<Context, Args...>()
+                : -static_cast<int64_t>(NUM_ARGS);
```

Comment:
Please factor out the initializer into a separate function and reuse here and below to reduce copy-paste.